### PR TITLE
GCW-2840-Edit subform Feature

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-2840

### What does this PR do?

FEATURE: Additional fields editing package type for packages already created. There are different scenarios for subform is handled in this PR.

### Impacted Areas

Package details page where user can edit existing subform to same/other/no subforms.

### Mockup Link

https://xd.adobe.com/view/224f7314-f1c0-40b6-7d7a-929e616be6bb-547e/screen/7d129040-9142-4ed1-ba71-8011197fae58/Create-Item-Computer-form/?x_product=xd-jiraserver/1.0

### CODE REVIEW

File: app/models/concerns/subform_utilities.rb [hunk 1]
```diff
@@ -29,6 +29,11 @@ def update_on_stockit
     add_stockit_id(response)
   end
 
+  def delete_on_stockit
+    return if request_from_stockit?
+    response = Stockit::ItemDetailSync.destroy(self)
```
Comment (abhinav098) [inline app/models/concerns/subform_utilities.rb]: setting it to response is not required @abulsayyad123 .

File: app/services/stockit/item_detail_sync.rb [hunk 2]
```diff
@@ -35,6 +39,11 @@ def update
       put(url, detail_params)
     end
 
+    def delete
+      url = url_for("/api/v1/#{detail_type.pluralize}/destroy")
```
Comment (abhinav098) [inline app/services/stockit/item_detail_sync.rb]:
```suggestion
      url = url_for("/api/v1/#{ detail_type.pluralize }/destroy")
```

File: app/services/stockit/item_detail_sync.rb [hunk 3]
```diff
@@ -35,6 +39,11 @@ def update
...
+      put(url, { detail_type => { 'id' => detail.stockit_id, 'item_id' => detail&.package&.stockit_id }})
```
Comment (abhinav098) [inline app/services/stockit/item_detail_sync.rb]:
```suggestion
      put(url, { detail_type => { 'id' => detail.stockit_id, 'item_id' => detail&.package&.stockit_id } })
```

**Comment by patrixr** [comment]
> * [x]  General Code Review
> * [x]  Functional Code Review
> * [x]  Language Specific Code Review
> * [x]  Performance
> * [x]  Test Cases Coverage
> 
> 👍

Loving the checklist :D

File: app/controllers/api/v1/computer_accessories_controller.rb [hunk 4]
```diff
@@ -42,6 +42,11 @@ def update
         update_and_render_object_with_errors(@computer_accessory)
       end
 
+      def destroy
```
Comment (patrixr) [inline app/controllers/api/v1/computer_accessories_controller.rb]: Side note: Those controllers look alike, definite refactoring potential :)

File: app/controllers/api/v1/packages_controller.rb [hunk 5]
```diff
@@ -89,6 +89,7 @@ def create
       param_group :package
 
       def update
+        @package.detail = assign_detail if params["package"]["detail_type"].present?
```
Comment (patrixr) [inline app/controllers/api/v1/packages_controller.rb]: won't it break if the payload doesn't have a `'package'` key ?

**Pull request merged**